pkg/envoy/resources/generator.go

```diff
@@ -0,0 +1,25 @@
+package envoy
+
+import (
+	"github.com/3scale/marin3r/pkg/envoy"
+	envoy_resources_v2 "github.com/3scale/marin3r/pkg/envoy/resources/v2"
+	envoy_resources_v3 "github.com/3scale/marin3r/pkg/envoy/resources/v3"
+)
+
+// Generator in an interface with methods to generate
+// envoy resource structs
+type Generator interface {
+	New(rType envoy.Type) envoy.Resource
+	NewSecret(string, string, string) envoy.Resource
+}
```

Comment:
This abstracts the generation of envoy resources from the version. Any envoy API version should implement this interface. Adding new API versions should be easier in the future.